components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManagerTest.java

```diff
@@ -0,0 +1,228 @@
+import static org.mockito.Matchers.anyMap;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+/*
+ * Unit tests for SCIMUserManager
+ */
+@PrepareForTest({SCIMGroupHandler.class, IdentityUtil.class, SCIMUserSchemaExtensionBuilder.class, SCIMAttributeSchema.class, AttributeMapper.class})
+@PowerMockIgnore("java.sql.*")
+public class SCIMUserManagerTest extends PowerMockTestCase {
+
+    @Mock
+    private UserStoreManager mockedUserStoreManager;
+
+    @Mock
+    private ClaimManager mockedClaimManager;
+
+    @Mock
+    private GroupDAO mockedGroupDAO;
+
+    @Mock
+    private SCIMAttributeSchema mockedScimAttributeSchema;
+
+    @Mock
+    private RealmConfiguration mockedRealmConfig;
+
+    @Mock
+    private User mockedUser;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+
+        initMocks(this);
+    }
+
+    @DataProvider(name = "ClaimData")
+    public Object[][] data() {
+
+        String CoreClaimUri1;
+        String testMappedAttributesCore1;
+
+        String CoreClaimUri2;
+        String testMappedAttributesCore2;
+
+        String UserClaimUri1;
+        String testMappedAttributesUser1;
+
+        String UserClaimUri2;
+        String testMappedAttributesUser2;
+
+        Claim claim1 = new Claim();
+        Claim claim2 = new Claim();
+        Claim claim3 = new Claim();
+        Claim claim4 = new Claim();
+
+        CoreClaimUri1 = "testCoreClaimURI1";
+        claim1.setClaimUri(CoreClaimUri1);
+        testMappedAttributesCore1 = "MappedAttributesCore1";
+
+        CoreClaimUri2 = "testCoreClaimURI2";
+        claim2.setClaimUri(CoreClaimUri2);
+        testMappedAttributesCore2 = "MappedAttributesCore2";
+
+        UserClaimUri1 = "testUserClaimURI1";
+        claim3.setClaimUri(UserClaimUri1);
+        testMappedAttributesUser1 = "MappedAttributesUser1";
+
+        UserClaimUri2 = "testUserClaimURI2";
+        claim4.setClaimUri(UserClaimUri2);
+        testMappedAttributesUser2 = "MappedAttributesUser2";
+
+        ClaimMapping cMap1 = new ClaimMapping(claim1, testMappedAttributesCore1);
+        ClaimMapping cMap2 = new ClaimMapping(claim2, testMappedAttributesCore2);
+        ClaimMapping cMap3 = new ClaimMapping(claim3, testMappedAttributesUser1);
+        ClaimMapping cMap4 = new ClaimMapping(claim4, testMappedAttributesUser2);
+
+        ClaimMapping[] coreClaims = new ClaimMapping[]{cMap1, cMap2, cMap3, cMap4};
+
+        HashMap<String, Boolean> requiredAttributes = new HashMap<String, Boolean>() {
+            {
+                put("test1.test", true);
+            }
+        };
```

Comment:
In here, you have extended the HashMap class, isn't it? I'm having trouble interpreting what you are trying to do here with double curly brackets. Raising this question for my understanding. :)

Reply:
I  actually didn't extend the HashMAp class rather I assigned values to the hashmap at the initialization. :)

Reply:
I disagree.. :D After much thought, I figured out the complexity here. You have created an anonymous class, and then added a instance-initializer block. You are calling the #put method inside that instance initializer block.

Isn't it easier to read if we simply created a hashmap instance, and then call its put method?